File: src/home/1145/main.py
import sys
sys.stdin = open('data.in', 'r')
sys.stdout = open('data.out', 'w')
names=[]

def main():
    n,t,s = map(int, input().split())
    for _ in range(n):
        line = input().strip()
        parts = line.split(',')
        attacker, weapon = parts[0].split('use')
        parts = parts[1:]
        weapon = weapon.upper()
        if 'AK' not in weapon:
            continue
        parts[0]=parts[0][4:]
        parts[-1]=parts[-1][:-1]
        for part in parts:
            if 'ash' == part:
                t=t-s
                names.append(attacker)
                break
    if t<=0:
        print("Heros never die.")
    else:
        print(len(names))
        for name in names:
            print(name)
if __name__ == '__main__':
    main()

File: src/home/1145/main.cpp
#define LOCAL
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"
int n, t, s;
vector<string> names;
signed main()
{
    cin.tie(0);
    cout.tie(0);
    ios::sync_with_stdio(0);
#ifdef LOCAL
    freopen("data.in", "r", stdin);
    freopen("data.out", "w", stdout);
#endif
    cin >> n >> t >> s;
    cin.ignore();
    while (n--)
    {
        string info;
        char ch;
        while (cin.get(ch) && ch != '\n')
        {
            info += ch;
        }
        // cout << info << endl;
        string name;
        size_t posName = info.find("use");
        name = info.substr(0, posName - 1);
        // cout << info << endl;
        size_t posUse = info.find("use") + 4;
        size_t firstComma = info.find(',');
        string Weapon = info.substr(posUse, firstComma - posUse - 1);
        transform(Weapon.begin(), Weapon.end(), Weapon.begin(), ::toupper);
        // cout << Weapon << endl;
        bool hited = Weapon.find("AK") != string::npos;
        size_t posHit = info.find("hit");
        if (!hited)
        {
            continue;
        }
        size_t pos = posHit;
        while (true)
        {
            size_t nextComma = info.find(',', pos + 1);
            if (nextComma == string::npos)
            {
                nextComma = info.find('.', pos + 1);
                if (nextComma == string::npos)
                {
                    break;
                }
            }
            string segment = info.substr(pos + 4, nextComma - pos - 4);
            if (segment == "ash")
            {
                t -= s;
                names.push_back(name);
                break;
            }
            pos = nextComma;
        }
    }
    if (t <= 0)
    {
        cout << "Heros never die." << endl;
    }
    else
    {
        cout << names.size() << endl;
        for (auto name : names)
        {
            cout << name << endl;
        }
    }
    return 0;
}